patches/tModLoader/Terraria/Recipe.TML.cs

```diff
@@ -426,12 +438,13 @@ public static class ConsumptionRules
 			return this;
 		}
 
-		internal static Recipe Create(Mod mod, int result, int amount) {
-			var recipe = new Recipe(mod);
-
+		public static Recipe Create(int result, int amount = 1) {
 			if (!RecipeLoader.setupRecipes)
 				throw new RecipeException("A Recipe can only be created inside recipe related methods");
 
+			ArgumentNullException.ThrowIfNull(RecipeLoader.CurrentMod);
```

Comment:
I wonder if these checks should be going into the Recipe ctor instead, wouldn't that be cleaner? ( I don't remember if there is a usecase for passing null for the mod )

Reply:
Vanilla recipes use `null` for mod